### ISSUE TITLE
Maybe change name (away from LSIO)

### ISSUE THREAD

**Issue by JackKelly** [JackKelly/light-speed-io#110]
Problems with the name "light speed IO"
- This project will ultimately be more about _compute_ than IO!
- In lowercase, the first character of `lsio` is ambiguous (is it a lowercase `L`, or a `1`, and an uppercase `I`?!

None of these problems are deal-breakers. So maybe I should just stick with LSIO.

## Thoughts about a better name
- Maybe something about processing data in chunks?
  - `Chunker` has the right meaning, but it's an ugly word, and sounds like 90's slang for throwing up!
- The main common theme is processing multi-dimensional, labelled arrays that are too large to fit into RAM, and to do so efficiently as possible on modern hardware.

Any thoughts?

**Comment by JackKelly** [comment]
Possible alternative names (these are all from Vincent Immler! Thanks Vincent!):

- warpspeedio   (because it is fast)
- wormholeio (gets you from one place to another without long travel)
- gigaio  (haha ...)
- octopusio or krakenio (its tentacles representing the many async threads)
- overdriveio   (put your stuff on overdrive!)
- hyperspeedio (hmm ... tooo lengthy)
- tachyonio   (I like this one and it is very unique)  [1]
- rexio (because it rules)
- maximusio (because it maximizes throughput)
- summitio
- pinnacleio
- spearheadio
- the goal is, we want to be faster than just light. "just as fast as light – booooring" 😋

**Comment by braddf** [comment]
How about `hopper-io`, after [Grace Hopper](https://en.wikipedia.org/wiki/Grace_Hopper) – credited inventor of the compiler? 
Not sure if I understand the gubbins of what you're doing and how enough, but "hopping" could be also a fun verb for input-output in a super quick way! Can even imagine the name being "genericised" and becoming verb ("to hop"/"to hopp") if this did become a widely-used library, as is the hope...

**Comment by braddf** [comment]
Also I guess it makes sense as a method/tool for delivering things, as in a grain hopper etc. – looks like `hopper` is a crate already, but doing something that sounds very different and probably not easily confused? But, again, I am not a Ruster. 😅

**Comment by JackKelly** [comment]
> How about hopper-io, after [Grace Hopper](https://en.wikipedia.org/wiki/Grace_Hopper)

Oooh, that's a nice idea! Thanks so much for the suggestion!

I must admit that I'm trying to move away from names with "IO" in them, because - ultimately - I'm expecting this project to be just as much about efficiently and quickly _computing_ multi-dimensional arrays as it's about _IO_!

Also, "Hopper" is the name of [Nvidia's previous-generation GPU architecture](https://en.wikipedia.org/wiki/Hopper_(microarchitecture))!

**Comment by devsjc** [comment]
I always had the concern as well with `lightspeed_io` that `lsio` is already the short name for a popular provider of docker containers [linuxserver.io](https://www.linuxserver.io/) - first thing on my google when I search "lsio" (but I'm perhaps entering with a biased algorithm!) - what with it being reasonable overlap in domain (software) I think another good reason to consider different names. 

My two pence: short and snappy takes precedence. `farr`, `roar` I like, and actually pretty partial to 👂`ears`, I think you could bump that up the list! Appreciate it would be hard to search but so often I have to [put a programming language qualifier after a tool search](https://www.google.com/search?q=ruff) I don't think that's a problem!

So I'm not a completely useless armchair commenter here's a couple more ideas, although I don't think they're better than any of yours already:
- 🚀 `scram` - suggests speed due to the [scramjet](https://en.wikipedia.org/wiki/Scramjet), could stand for something like *Speedy out-of-Core (Random Access) Memory array processor*. 🚀 "*Tell your slow compute tasks to `scram`*!"
- 💪 `'ard` - e.g. *Array data processor*.  "*Make 'ard tasks easy*!" Maybe not enough in the name for the magnitude of the task and might be weird as a prefix.
- 🙂 `mdma` - suggests speed due to...! Could stand for something like *Multi-Dimensional out-of-core Memory Array processor, and the taglines make themselves up!

Loving following the work so far Jack!

(Also, to think all this time I thought Fender guitars was spelt with an "er"...! 😉)

**Comment by JackKelly** [comment]
Thanks @devsjc - these are great comments and suggestions! Thank you for flagging up "LinuxServer.io". I agree: that's _so_ similar that I think I definitely do need to change the name!

**Comment by JackKelly** [comment]
After "sleeping on it" over the weekend, I think I'm going to opt for `farr` (short for fast array). I'll admit that I'm not wildly in love the name! But it makes logical sense, it's concise, and I'm not aware of any deal-breakers.

If you have better suggestions, or if you can see a fatal flaw in the name `farr` then please shout now! If not, I'll probably rename the project to `farr` early next week

**Comment by braddf** [comment]
Random thought, looks like you've resolved this thread, but I'll put here anyway – how about `sting`? 
- Fast in and out;
- Available on cargo with no similar ones I can see?
- Nice wordplay in combo with array (if slightly reminiscent of a native Japanese speaker with a strongly accented pronunciation of the marine animal) – which could double as a logo/icon idea.

**Comment by braddf** [comment]
Or another very different one – `corten`, after the steel alloy(?) which features an intentional, protective layer of `rust`, because this is the most dependable way to not develop rust in an uncontrolled manner when you build something, and as it will probably end up having rust somewhere (maybe everywhere) at some point, it's best to start with intentional, well-planned and well-founded thin layer of `rust` from the very start of the process. 

The metaphor is in there somewhere.

**Comment by JackKelly** [comment]
Ooh, nice suggestions!

`corten` is cool! (Also, I honestly only just realised that it's spelled "corten"! I'd heard "corten" said on Grand Designs, but I always thought that Kevin was saying "core 10"!)

There is an existing [crate called `corten`](https://crates.io/crates/corten) but the description of the crate says "_Rust is good for metal. Just reserving a cool name. Willing to transfer ownership if requested._"

If we can think of a way to link `corten` to processing large arrays (!) then that'd be awesome :slightly_smiling_face:... something about `corten` being strong and beautiful, whilst also working _with_ the environmental conditions.

**Comment by JackKelly** [comment]
Although, that said, plenty of software packages have names which have absolutely no "inner meaning"... [`ruff`](https://github.com/astral-sh/ruff), `pandas`,  [`polars`](https://crates.io/crates/polars), [`tokio`](https://crates.io/crates/tokio), etc.

**Comment by braddf** [comment]
Yeah I didn't know what it meant for an embarrassing amount of time too 😂 if your googling hasn't already pointed this out, I'm pretty sure it's an elision/stands for its two main properties, "corrosion resistance" and "tensile strength"... 

It is a structural material, often in beams, which simplistically are 1-dimensional (or at least single axis), so in a multidimensional way of putting stuff together, (up to 3 at least) it stacks up... 👍 haha

**Comment by JackKelly** [comment]
In the last few days, several people have commented how much the like the name "light speed IO".

I'm not sure what's best to do with the name: leave it as is, or change it!

**Comment by JackKelly** [comment]
So... we probably do need to move away from `LSIO`... but some folks do like the "light speed" part of the name.

Maybe we can keep "light speed" in the name? Some options:

- `light-speed-array`: `LSA`. 
    - Both `light-speed-array` and `LSA` are available on Crates.io.
    - Sub-crate names sound OK: `lsa_io`, `lsa_object_store_bridge`, `lsa_uring`.
    - Still has the problem that the lowercase "L" is ambiguous in some fonts.
- Use `c` in the acronym, so `light-speed-array` becomes `c-array` (although that might be confusing with the programming language `C`!)
- `light-speed-data`: has an unfortunate acronym!
